scripts/generate_checksum.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+SHASUM_FILES="shasums-eksctl-anywhere"
+
+#Clean up any existing old folders for shasum files
+if [ -d "$SHASUM_FILES" ]; then rm -Rf $SHASUM_FILES; fi
+
+YQ_LATEST_RELEASE_URL="https://github.com/mikefarah/yq/releases/latest"
+
+if ! command -v yq &> /dev/null
+then
+    echo "Please install YQ from $YQ_LATEST_RELEASE_URL and rerun this script."
+    exit
+fi
+
+echo "YQ is installed, continuing.."
+
+FILE_PREFIX="eksctl-anywhere"
+
+DARWIN_TAR="darwin-amd64.tar.gz"
+LINUX_TAR="linux-amd64.tar.gz"
+
+CURRENT_YQ_VERSION=$(yq -V | awk '{print $NF}')
+
+versionSplit=($(echo $CURRENT_YQ_VERSION | tr "." "\n"))
+CURRENT_YQ_MAJOR_VERSION=${versionSplit[0]}
+CURRENT_YQ_MINOR_VERSION=${versionSplit[1]}
+EXPECTED_YQ_MAJOR_VERSION=4
+EXPECTED_YQ_MINOR_VERSION=18
+
+if [ $(($CURRENT_YQ_MAJOR_VERSION)) -lt $(($EXPECTED_YQ_MAJOR_VERSION)) ]; then
+    echo "ERROR: Current yq major version v$CURRENT_YQ_MAJOR_VERSION.x is older than expected v$EXPECTED_YQ_MAJOR_VERSION.x"
+    echo "Please install the latest version of yq from $YQ_LATEST_RELEASE_URL"
+    exit 1
+fi
+
+if [ $CURRENT_YQ_MAJOR_VERSION == $(($EXPECTED_YQ_MAJOR_VERSION)) ] && [ $CURRENT_YQ_MINOR_VERSION -lt $(($EXPECTED_YQ_MINOR_VERSION)) ]; then
```

Comment:
nit: since the check on line 43 already compares the major versions, I think we won't need to compare `if [ $CURRENT_YQ_MAJOR_VERSION == $(($EXPECTED_YQ_MAJOR_VERSION)) ]` again right?

Reply:
we are just checking for less than above for the major versions, but say we have a 5.1 comparing to 4.18, the minor versions should be checked only when the major version is 4, and nothing else.